rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -270,6 +270,69 @@ def __init__(self,
         super(ROCALClassificationIterator, self).__init__(pipe, tensor_layout=pipe._tensor_layout, tensor_dtype=pipe._tensor_dtype,
                                                           multiplier=pipe._multiplier, offset=pipe._offset, display=display, device=device, device_id=device_id)
 
+class ROCALAudioIterator(object):
+    """! ROCAL iterator for audio tasks for PyTorch
+    The Tensors that are returned by the iterator will be owned by ROCAL and would be valid until next iteration.
+        @param pipeline            The rocAL pipeline to use for processing data.
+        @param tensor_dtype        Data type of the output tensors.
+        @param size                      Number of samples in the epoch (Usually the size of the dataset).
+        @param auto_reset                Whether the iterator resets itself for the next epoch or it requires reset() to be called separately.
+        @param device              The device to use for processing
+        @param device_id           The ID of the device to use
+    """
+    def __init__(self, pipeline, tensor_dtype = types.FLOAT, size = -1, auto_reset = False, device = "cpu", device_id = 0):
+        self.loader = pipeline
+        self.device = device
+        self.device_id = device_id
+        self.output = None
+        self.iterator_length = b.getRemainingImages(self.loader._handle) # To change the name of getRemainingImages to getRemainingSamples in upcoming PRs
+        self.max_shape = None
+        self.batch_size = self.loader._batch_size
+        self.output_list = None
+        self.labels_size = self.batch_size
+        self.output_memory_type = self.loader._output_memory_type
+
+    def next(self):
+        return self.__next__()
+
+    def __next__(self):
+        if self.loader.rocal_run() != 0:
+            raise StopIteration
+        else:
+            self.output_tensor_list = self.loader.get_output_tensors()
+
+        self.output_list = []
+        for i in range(len(self.output_tensor_list)):
+            dimensions = self.output_tensor_list[i].dimensions()
+            self.num_roi_dims = self.output_tensor_list[i].roi_dims_size()
+            self.roi_array = np.zeros(self.batch_size * self.num_roi_dims * 2, dtype=np.int32)
+            self.output_tensor_list[i].copy_roi(self.roi_array)
+            if self.device == "cpu":
+                torch_dtype = self.output_tensor_list[i].dtype()
```

Comment:
torch_dtype is common for both cpu and gpu ryt? Can we move it before the if condition so only 1 variable will be declared?